libp2p/security/base_session.py

```diff
@@ -31,7 +31,7 @@ def __init__(
         self.remote_permanent_pubkey = None
 
         self.conn = conn
-        self.initiator = self.conn.initiator
+        self.initiator = peer_id is not None
```

Comment:
Since `initiator` is still exposed in `IRawConnection`, why is the reason we change to `peer_id is not None`?

Reply:
i ended up removing the `conn` as an argument here in a later PR (and also what happened was that I was passing in a `MsgIOReadWriter` that doesn't have the `initiator` property... something feels pretty ad-hoc to me about attaching the `initiator` flag to the connection we pass around so i'm generally working to remove it)